src/core_ocean/shared/mpas_ocn_adcReconstruct.F

```diff
@@ -445,6 +470,9 @@ subroutine compute_ADC_tends(nCells,nVertLevels, nTracers, dt,activeTracers, uve
                 SwD = -(1.0_RKIND-2.0_RKIND*areaFraction(k+1,iCell)) &
                    /(areaFraction(k+1,iCell)*(1.0_RKIND-areaFraction(k+1,iCell)))**0.5_RKIND
 
+                Ustz = (uStokes(k,iCell) - uStokes(k+1,iCell)) / dz
+                Vstz = (vStokes(k,iCell) - vStokes(k+1,iCell)) / dz
+
```

Comment:
these terms are using dz = ze(k) - ze(k+1), really should use `zm(k) - zm(k+1)`.  Won't affect our testing now, but would be good to change

Reply:
Also this derivative seems to live at an interface, but we need this to live at the center, so something like
Ustz = (ustokes(k-1) - ustokes(k+1)) / (zm(k-1) - zm(k+1))

but perhaps I'm misunderstanding

Reply:
Great catch! This derivative exists on cell centers. This is correct for the w3 budget, but for w2 and other second-moments it needs to exist on interfaces. I think I have now fixed this by calculating Stokes drifts on all edge and interface points, and calculating their derivatives at cell centers and interfaces for w3 and second-moment budgets respectively.